### PR TITLE
feat(tools): husky, commitlint and commitizen was added

### PR DESCRIPTION
**Pull request by ValeraS**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [ ] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [ ] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [ ] Closes currently open issue: Closes #XXX

#### Description

- commitizen friendly repository is made as in the freecodecamp/curriculum
- added lint checks on the commit message
- added lint project checks on the pre-commit hook.

### CODE REVIEW

**Comment by tchaffee** [comment]
@ValeraS Thanks for the PR. Let's check with the core team about what they are doing so we keep it consistent from repo to repo.

@QuincyLarson @raisedadead @Bouncey do we plan on continuing to use husky and commitizen with the other freeCodeCamp projects? I notice husky and commitizen are used on the freeCodeCamp main repo but not on the learn repo? Are we moving away from those tools?

**Comment by QuincyLarson** [comment]
@raisedadead @Bouncey what are your thoughts on Husky? Are we planning to keep it? If so, I agree that we should use it on all our main repos.

**Comment by Bouncey** [comment]
I like what they do for us, but i dont like it when the requurements are
too harsh. If we expand it out to all repos, the settings should remain the
same.

@raisedadead is there scope to to have things like the commit message rules
in an npm package, much like our eslint config

**Comment by ValeraS** [comment]
@raisedadead Should we use https://github.com/freeCodeCamp/cz-freecodecamp somehow or all repos do a self-config?

Should I also add @commitlint/travis-cli and a command in .travis.yml?

**Comment by tchaffee** [comment]
@ValeraS @raisedadead @Bouncey @QuincyLarson my recommendation is to merge this PR and adopt the best practices agreed on. For now we can copy the standards from freeCodeCamp main repo and @raisedadead should open a new issue in the main repo to put the rules in their own repo to be shared across all repos and maintained in one place. That's a related secondary goal but not strictly necessary for this PR. In the meantime, we all have to remember to communicate with each other about any changes to the freeCodeCamp main repo automated commit rules. That should be the source of truth, but we will need to propagate any changes manually for now.

**Comment by raisedadead** [comment]
@ValeraS I think `cz-freecodecamp` needs to be updated. I would recommend taking a look at https://github.com/freeCodeCamp/curriculum which is already configured with the setup that we need.

**Comment by raisedadead** [comment]
@ValeraS once you have cross referenced the configuration on the curriculum and updated this PR for any changes,  we are ready to merge this.

@tchaffee yes. That is an ideal place to keep our config together. I'll do the needful

File: commitizen.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+];
+
+const scopes = ['tests', 'tools', 'scripts'];
+
+module.exports = {
+  types,
+  scopes,
+  allowCustomScopes: true,
+  allowBreakingChanges: ['feat', 'fix', 'perf', 'refactor']
```
Comment (tchaffee) [inline commitizen.config.js]: Is refactor a valid type? I don't see it in the list above?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+node_modules
+build
+local_test
```
Comment (tchaffee) [inline .eslintignore]: This line should have a newline.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: commitizen.config.js (hunk 1 above)
Comment (ValeraS) [reply commitizen.config.js]: @tchaffee it's how it is in curriculum repo

Delete it or add it to types?

Comment (tchaffee) [reply commitizen.config.js]: That helps me understand where this came from. Let's leave everything as close to the curriculum repo as possible and only move away from that if it starts to cause us problems. The eventual goal is to share one standard.

File: .eslintignore (hunk 2 above)
Comment (tchaffee) [reply .eslintignore]: @ValeraS If you can just fix this one line I am ready to merge this. Thanks.

Comment (ValeraS) [reply .eslintignore]: @tchaffee it's done.

**Pull request merged**